MDNetworking/MDGameSession.cs

```diff
@@ -410,7 +430,26 @@ private void ClientToServerCommand(string Command)
         #endif
     }
 
-    public Node SpawnNetworkedNode(Type NodeType, Node Parent, string NodeName, int NetworkMaster = -1, Vector3? SpawnPos = null)
+    /// <summary>Returns adds a random number to the name if the second boolean parameter is true</summary>
+    protected String GetNodeName(String Name, bool UseRandomName)
+    {
+        if (UseRandomName)
+        {
+            // Call ourselves again to guarantee that the new name is unique
```

Comment:
I think instead of a random number, either Object.GetInstanceId() or Guid.NewGuid() should be used as they're a bit more appropriate for this use.